db/tables.yml

```diff
@@ -192,12 +186,13 @@ task_groups:
   expires: timestamp with time zone not null
   scheduler_id: text not null
   task_group_id: text not null
-queue_task_dependency_entities:
-  partition_key: text not null
-  row_key: text not null
-  value: jsonb not null
-  version: integer not null
-  etag: uuid
+task_dependencies:
+  dependent_task_id: text not null
+  required_task_id: text not null
+  expires: timestamp with time zone not null
+  requires: task_requires not null
```

Comment:
It looks like we could add another type with [ALTER TYPE](https://www.postgresql.org/docs/9.1/sql-altertype.html).

Reply:
Will `ALTER TYPE` on a large table require the upgrade to happen during a TCW (i.e, takes a long time to execute)?

Reply:
It only allows *adding* an enum value and not removing, so I'm assuming it doesn't touch the tables at all - it's just adding an additional valid value, without changing any of the existing values.

Reply:
Got it. How will we handle removing in this case if we use db types?

Reply:
That would be a breaking API change, I think, and we'd have to have some way of ensuring that all tasks did not have the old value to be removed before removing it.  In all likelihood, we'd deprecate the old value but leave it in place indefinitely.

Reply:
I see. The reason I was asking is because I vaguely remember we had a discussion about whether to pick `varchar` or `text` for data types and I think we agreed to keep the constraints at the api schema level rather than the db. Anyways, the enum change makes sense to me but I just wanted to make sure I understand what is acceptable to do in the db layer and what is better left for the schema layer to take care of so that we are consistent with all the other db tables.

Reply:
That makes sense.  My reasoning to use an enum here is that since this is always one of two strings, there's no need to store tens of millions of copies of those strings.  Enums are integers, so they fit in a few bytes.

Reply:
That makes a lot of sense. Thank you Dustin.